app/models/collection_preference.rb

```diff
@@ -1,15 +1,31 @@
 class CollectionPreference < ApplicationRecord
   belongs_to :collection
+
+  after_create :set_initial_audit_timestamps
+  after_update :set_audit_timestamps
   after_update :after_update
 
+  private
+
+  def set_initial_audit_timestamps
+    update_columns(
+      unrevealed_updated_at: created_at,
+      anonymous_updated_at: created_at
+    )
+  end
+
+  def set_audit_timestamps
```

Comment:
audit means something specific for us internally, so I'd prefer to not use that word here. Maybe `set_updated_at_timestamps` or something similar?